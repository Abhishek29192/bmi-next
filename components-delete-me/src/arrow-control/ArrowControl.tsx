import React from "react";
import classnames from "classnames";
import { KeyboardArrowRight } from "@material-ui/icons";
import { KeyboardArrowLeft } from "@material-ui/icons";
import Button from "../button/Button";
import styles from "./ArrowControl.module.scss";

type Props = {
  direction: "left" | "right";
  onClick?: () => void;
};

const ArrowControl = ({ direction, onClick }: Props) => {
  // eslint-disable-next-line security/detect-object-injection
  const ArrowIcon = {
    left: KeyboardArrowLeft,
    right: KeyboardArrowRight
  }[direction];

  return (
    <Button
      isIconButton
      size="medium"
      onClick={() => onClick && onClick()}
      className={classnames(
        styles["ArrowControl"],
        styles[`ArrowControl--${direction}`]
      )}
      // TODO: Use a UI string for this.
      accessibilityLabel={`Move ${direction}`}
    >
      <ArrowIcon className={styles["chevron"]} />
    </Button>
  );
};

export default ArrowControl;
