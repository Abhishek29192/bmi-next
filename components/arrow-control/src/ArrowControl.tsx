import React from "react";
import Button from "@bmi/button";
import classnames from "classnames";
import ChevronRightIcon from "@material-ui/icons/KeyboardArrowRight";
import ChevronLeftIcon from "@material-ui/icons/KeyboardArrowLeft";
import styles from "./ArrowControl.module.scss";

type Props = {
  direction: "left" | "right";
  onClick?: () => void;
};

const ArrowControl = ({ direction, onClick }: Props) => {
  // eslint-disable-next-line security/detect-object-injection
  const ArrowIcon = {
    left: ChevronLeftIcon,
    right: ChevronRightIcon
  }[direction];

  return (
    <Button
      isIconButton
      size="medium"
      onClick={() => onClick && onClick()}
      className={classnames(
        styles["ArrowControl"],
        styles[`ArrowControl--${direction}`]
      )}
      // TODO: Use a UI string for this.
      accessibilityLabel={`Move ${direction}`}
    >
      <ArrowIcon className={styles["chevron"]} />
    </Button>
  );
};

export default ArrowControl;
