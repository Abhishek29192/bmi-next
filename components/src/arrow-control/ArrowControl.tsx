import { KeyboardArrowLeft, KeyboardArrowRight } from "@material-ui/icons";
import classnames from "classnames";
import React from "react";
import Button from "../button/Button";
import { useStyles } from "./styles";

type Props = {
  direction: "left" | "right";
  onClick?: () => void;
};

const ArrowControl = ({ direction, onClick }: Props) => {
  const classes = useStyles();
  // eslint-disable-next-line security/detect-object-injection
  const ArrowIcon = {
    left: KeyboardArrowLeft,
    right: KeyboardArrowRight
  }[direction];

  return (
    <Button
      isIconButton
      size="medium"
      onClick={() => onClick && onClick()}
      /* eslint-disable-next-line security/detect-object-injection */
      className={classnames(classes.root, classes[direction])}
      // TODO: Use a UI string for this.
      accessibilityLabel={`Move ${direction}`}
    >
      <ArrowIcon className={classes.chevron} />
    </Button>
  );
};

export default ArrowControl;
