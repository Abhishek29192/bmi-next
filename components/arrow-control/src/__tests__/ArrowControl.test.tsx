import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ArrowControl from "../";

describe("ArrowControl component", () => {
  it("renders correctly", () => {
    const { container } = render(<ArrowControl direction="left" />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("triggers an onClick event", () => {
    const onClick = jest.fn();
    const { getByLabelText } = render(
      <ArrowControl direction="left" onClick={onClick} />
    );

    // TODO: Change this when the string becomes dynamic.
    fireEvent.click(getByLabelText("Move left"));
    expect(onClick.mock.calls).toMatchSnapshot();
  });
});
