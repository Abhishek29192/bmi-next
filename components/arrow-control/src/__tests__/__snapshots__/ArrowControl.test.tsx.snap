// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowControl component renders correctly 1`] = `
<button
  aria-label="Move left"
  class="MuiButtonBase-root MuiIconButton-root iconButtonStyles-root-11 iconButtonStyles-medium-17 iconButtonStyles-contained-12 ArrowControl ArrowControl--left"
  tabindex="0"
  type="button"
>
  <span
    class="MuiIconButton-label"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root chevron"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
      />
    </svg>
  </span>
  <span
    class="MuiTouchRipple-root"
  />
</button>
`;

exports[`ArrowControl component triggers an onClick event 1`] = `
Array [
  Array [],
]
`;
